File: Mode_Benchmark/submit.py
import pandas as pd

test = pd.read_csv("test.csv")
nrow = test.shape[0]
output = []
for i in range(0, nrow):
	print i, nrow
	y = (test.Sequence[i].split(","))
	y = map(int, y)
	output.append(max(set(y), key=y.count))

submit = pd.read_csv("sample_submission.csv")
submit["Last"] = output
submit.to_csv("output.csv", index=False)

File: Mode_Benchmark/submit.r
library(dplyr)
options(scipen=999)

Mode <- function(x){
	ux <- unique(x)
	return(ux[which.max(tabulate(match(x, ux)))])
}

parse <- function(str){
	return(as.numeric(strsplit(str, split=",")[[1]]))
}

test = read.csv("test.csv", stringsAsFactors=FALSE)
# print(head(test))
# for(i in 1:nrow(test)){
# 	print(test$Sequence[i])
# 	print(Mode(parse(test$Sequence[i])))
# 	break
# }

# x = c(1,2,2,2)
# print(Mode(x))

test %>%
rowwise() %>%
print(Sequence) %>%
mutate(Last = Mode(parse(Sequence))) %>%
select(Id, Last) %>%
arrange(Id) %>%
write.csv(., "mode_benchmark.csv", row.names=FALSE)